lib/PuppeteerSharp/Connection.cs

```diff
@@ -245,26 +249,23 @@ private void ProcessIncomingMessage(ConnectionResponse obj)
             if (!string.IsNullOrEmpty(obj.SessionId))
             {
                 var session = GetSession(obj.SessionId);
-                session.OnMessage(obj);
+                session?.OnMessage(obj);
```

Comment:
When would this be null?

Reply:
This can technically be null since `_sessions.GetValueOrDefault(sessionId)` can return null, but I encountered this once during a test run, which seemed to happen at random.

Is this a possible path?
```csharp
            else if (method == "Target.detachedFromTarget")
            {
                // -> exec
                var sessionId = param.SessionId; // sessionId = "X"
                if (_sessions.TryRemove(sessionId, out var session) && !session.IsClosed)
                {
                    // -> exec
                    session.Close("Target.detachedFromTarget");
                }
            }

            if (!string.IsNullOrEmpty(obj.SessionId))
            {
                // -> exec
                var session = GetSession(obj.SessionId); // obj.SessionId = "X"
                // session = null since it was just removed above
                session?.OnMessage(obj);
            }
```

Not sure when/why `param.SessionId == obj.SessionId` though.